### ISSUE TITLE
Path Incorrect

### ISSUE THREAD

**Issue by mayurankv** [zjhcn/obsidian-code-preview#2]
The plugin says no such file or directory but the path given seems to give my vault path twice before the path I actually pass (instead of once). 

If I do absolute I get $VAULT_PATH/$VAULT_PATH/my_path/...
If I do relative I get $VAULT_PATH/$RELATIVE_PATH/my_path/...

**Comment by hankjs** [comment]
The cause of the problem is that ObsidianAPI only supports relative paths for read files.

I hope the following examples will be helpful to you.

---

If you files path like: 
- /home/user/vault/my_path/file.js
- /home/user/other_project/out.js

$VAULT_PATH is "/home/user/vault".

The YAML configuration should look like this:

## Absolute

Vault file

```preview
path: /my_path/file.js
```

Other Porject

```preview
path: /../other_project/out.js
```

## Relative

**Relative paths work only for files of a fixed level**

### /home/user/vault/Home.md

Vault file

```preview
path: my_path/file.js
```

Other Porject

```preview
path: ../../out.js
```

### /home/user/vault/my_path/Foo.md

Vault file

```preview
path: file.js
```

Other Porject

```preview
path: ../../../out.js
```

**Comment by mayurankv** [comment]
I understand unix path behaviour and agree with what you have written above but this doesn't work for me. What I was saying is that despite having a valid path, I get the error `ENOENT: no such file or directory, open ` and then a path which has my vaultpath specified twice whether I use absolute, relative or reverse out of folders with ../

An arbitrarily long line `../../../../../../../../../../`...`/../../../../Test.py` gives the error path couldn't find as VAULT_PATH/Test.py when I would expect it to be `/Test.py` the root directory. `/../../../../../../../../../../`...`/../../../../Test.py` gives the exact same issue as well.

**Comment by hankjs** [comment]
> I understand unix path behaviour and agree with what you have written above but this doesn't work for me. What I was saying is that despite (尽管) having a valid (有效) path, I get the error `ENOENT: no such file or directory, open ` and then a path which has my vaultpath specified (指定) twice whether (是否) I use absolute (绝对) , relative (相对) or reverse (逆向) out of folders with ../
> 
> An arbitrarily long (长) line `../../../../../../../../../../`...`/../../../../Test.py` gives the error path couldn't find as VAULT_PATH/Test.py when I would (会) expect (期望) it to be `/Test.py` the root (根) directory (目录) . `/../../../../../../../../../../`...`/../../../../Test.py` gives the exact (确切) same issue as well.

---

Fixed in 1.3.3. The reason is that my code doesn't work on UNIX.

**Comment by mayurankv** [comment]
Thank you! Look forward to testing it out!

**Comment by mayurankv** [comment]
Just want to reopen this issue with one more problem (should be a quick fix I believe).

The absolute path works perfectly as expected. The relative path though has a problem though (doesn't work right now). I presume it takes some VAULT_PATH variable and RELATIVE_PATH variable and combines them to get PATH=VAULT_PATH+RELATIVE_PATH. However, the VAULT_PATH ends in `.../foo/bar` without a trailing `/` and the RELATIVE_PATH begins with `baz/...` without a preceding `/` so when they combine it looks like `.../foo/barbaz/...` without a middle `/`. This doesn't allow it to parse the proper file.

**Comment by hankjs** [comment]
> Just want to reopen this issue with one more problem (should be a quick fix I believe).
> 
> The absolute (绝对) path works perfectly (完全) as expected (期望) . The relative (相对) path though (虽然) has a problem though (doesn't work right now). I presume (推定) it takes some VAULT_PATH variable and RELATIVE_PATH variable and combines (结合) them to get PATH=VAULT_PATH+RELATIVE_PATH. However, the VAULT_PATH ends in `.../foo/bar` without a trailing (踪迹) `/` and the RELATIVE_PATH begins with `baz/...` without a preceding (之前) `/` so when they combine (结合) it looks like `.../foo/barbaz/...` without a middle (中间) `/`. This doesn't allow it to parse the proper (适当) file.

---

Thank you for the case. Fixed in 1.3.4

**Comment by mayurankv** [comment]
Thanks!
